Framework/CurveFitting/test/Functions/AsymmetricPearsonVIITest.h

```diff
@@ -0,0 +1,596 @@
+        // get value and add to the vector
+        vec_jocob_deriv.emplace_back(jacobian.get(0, 4));
+        param_value += mr_resolution;
+      }
+
+      // calculate numerically
+      std::vector<double> vec_mr;
+      std::vector<double> vec_numeric_deriv;
+      numerical_param_partial_derivative(ap7, 4, min_mr, max_mr, mr_resolution, x, vec_mr, vec_numeric_deriv);
+
+      // compare
+      for (size_t i = 0; i < vec_mr.size(); ++i) {
+        TS_ASSERT_DELTA(vec_jocob_deriv[i], vec_numeric_deriv[i], 1.0e-2);
+      }
+    }
+    return;
+  }
+
+  /** Test limit ml-->0
+   * @brief testLeftShapeLimit: test if the left shape limit of the
+   * asymmetric Pearson VII function as ml->0 is properly evaluated
+   */
+  void testLeftShapeLimit() {
+    const double ph = 100.0;
+    const double pc = 0.0;
+    const double width = 0.7;
+    const double ml = 0.0;
+    const double mr = 2.0;
+    const double expected_pred = ph / 2.0;
+
+    IPeakFunction_sptr ap7 = getInitializedAPVII(ph, pc, width, ml, mr);
+    std::vector<double> xValues{-1.2};
+    FunctionDomain1DVector domain(xValues);
+    FunctionValues values_ap7(domain);
+    ap7->function(domain, values_ap7);
+
+    // check asymmetric Pearson VII at ml = 0
+    TS_ASSERT_DELTA(expected_pred, values_ap7[0], 1e-4);
+
+    Mantid::CurveFitting::Jacobian jacobian(domain.size(), 5);
+    std::vector<double> vec_deriv_pred;
+    double expected_deriv_pred[5] = {0.5, 0.0, 0.0, -123.2144, 0.0};
+    ap7->functionDeriv(domain, jacobian);
+
+    // check derivatives at ml = 0 and x = -1.2
+    for (size_t i = 0; i < 5; ++i) {
+      vec_deriv_pred.emplace_back(jacobian.get(0, i));
+      TS_ASSERT_DELTA(expected_deriv_pred[i], vec_deriv_pred[i], 1e-4);
+    }
+  }
+
+  /** Test limit mr-->0
+   * @brief testRightShapeLimit: test if the right shape limit of the
+   * asymmetric Pearson VII function as mr->0 is properly evaluated
+   */
+  void testRightShapeLimit() {
+    const double ph = 1.0;
+    const double pc = 10.0;
+    const double width = 1.7;
+    const double ml = 0.2;
+    const double mr = 0.0;
+    const double expected_pred = ph / 2.0;
+
+    IPeakFunction_sptr ap7 = getInitializedAPVII(ph, pc, width, ml, mr);
+    std::vector<double> xValues{12};
+    FunctionDomain1DVector domain(xValues);
+    FunctionValues values_ap7(domain);
+    ap7->function(domain, values_ap7);
+
+    // check asymmetric Pearson VII at mr = 0
+    TS_ASSERT_DELTA(expected_pred, values_ap7[0], 1e-4);
+
+    Mantid::CurveFitting::Jacobian jacobian(domain.size(), 5);
+    std::vector<double> vec_deriv_pred;
+    double expected_deriv_pred[5] = {0.5, 0.0, 0.0, 0.0, -0.855666};
+    ap7->functionDeriv(domain, jacobian);
+
+    // check derivatives at mr = 0 and x = 12
+    for (size_t i = 0; i < 5; ++i) {
+      vec_deriv_pred.emplace_back(jacobian.get(0, i));
+      TS_ASSERT_DELTA(expected_deriv_pred[i], vec_deriv_pred[i], 1e-4);
+    }
+  }
+
+  void test_with_Levenberg_Marquardt() {
```

Comment:
what is the purpose of this test?

Reply:
It checks if the fitting minimization process is stable. I adopted it from "GaussianTest.h"